### PR TITLE
NamespaceLifecycle admission control plugin

### PR DESCRIPTION
**Pull request by derekwaynecarr**
If a Namespace is undergoing graceful termination, prohibit any CREATE actions in that namespace.  This admission control plug-in will be enabled once we have a NamespaceController that is actually doing the graceful termination.

See #5195 for design details.

### CODE REVIEW

File: plugin/pkg/admission/namespace/lifecycle/admission.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+func NewLifecycle(c client.Interface) admission.Interface {
+	store := cache.NewStore(cache.MetaNamespaceKeyFunc)
+	reflector := cache.NewReflector(
+		&cache.ListWatch{
+			ListFunc: func() (runtime.Object, error) {
+				return c.Namespaces().List(labels.Everything())
+			},
+			WatchFunc: func(resourceVersion string) (watch.Interface, error) {
+				return c.Namespaces().Watch(labels.Everything(), labels.Everything(), resourceVersion)
```
Comment (bgrant0607) [inline plugin/pkg/admission/namespace/lifecycle/admission.go]: I believe we have a fields.Everything() now.

File: plugin/pkg/admission/namespace/lifecycle/admission.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+	if namespace.Status.Phase != api.NamespaceTerminating {
+		return nil
+	}
+
+	name := "Unknown"
+	obj := a.GetObject()
+	if obj != nil {
+		name, _ = meta.NewAccessor().Name(obj)
+	}
+	return apierrors.NewForbidden(kind, name, fmt.Errorf("Namespace %s is terminating", a.GetNamespace()))
```
Comment (bgrant0607) [inline plugin/pkg/admission/namespace/lifecycle/admission.go]: Forbidden seems reasonable.

**Comment by bgrant0607** [comment]
Assigning to @erictune, who I believe is more familiar with admission control plugins.

**Comment by bgrant0607** [comment]
LGTM

File: plugin/pkg/admission/namespace/lifecycle/admission.go (hunk 1 above)
Comment (bgrant0607) [inline plugin/pkg/admission/namespace/lifecycle/admission.go]: Ooops. Ping.

Comment (derekwaynecarr) [inline plugin/pkg/admission/namespace/lifecycle/admission.go]: Will update this.  My other PR that actually does the delete has the changes to move to fields.Everything()

**Comment by derekwaynecarr** [comment]
@bgrant0607 - made the update to move to fields.Everything() so this should be good if we get green travis

**Pull request merged**